### PR TITLE
[.NET upgrade] Upgrade .NET to 4.6

### PR DESCRIPTION
**Pull request by mcintyre94**
These two commits are analogous to #16 and #18.  

The first contains the .NET 4.6 web installer, and updates references to use it. The second tells MSBuild to use v4.6 and updates the patch CSProj diff files to build with v4.6. The tools version remains at 12.0 so there are less changed files than in #18.

### CODE REVIEW

**Comment by kc284** [comment]
The dotnet packages are also used by the SDKs. Before accepting this we need to check which windows and in particular powershell versions 4.6 is supported on. If it's the same as 4.5, ok. If the change forces users to upgrade from a powershell version still in LCM we may need to consider building a 4.5 and a 4.6 version.

**Comment by mcintyre94** [comment]
OS wise we can compare the list at https://www.microsoft.com/en-gb/download/details.aspx?id=42642 for 4.5.2 with the list at https://www.microsoft.com/en-gb/download/details.aspx?id=48130 for 4.6. Support is identical at 4.6 (Vista and Server 2008 SP2 are dropped in 4.6.1 onward)

The SDKs require Powershell 4.0+ (https://github.com/xapi-project/xen-api-sdk/blob/master/powershell/README.txt.dist), and I have tested running Powershell 4 with .net 4.6 using a Windows 7 VM, so I believe support here is identical too.

**Comment by kc284** [comment]
This looks ok to me; let's have a test build done on Jenkins to ensure the output is indeed the expected and then we can merge.

File: mk/dotnet-packages-build.sh [hunk 1]
```diff
@@ -141,6 +141,7 @@ run_msbuild_nofw()
 }
 
 cd ${SCRATCH_DIR}/xml-rpc.net/src && run_msbuild
+cd ${SCRATCH_DIR}/xml-rpc_v45.net/src && run_msbuild && mv ../bin/CookComputing.XmlRpcV2.dll ../bin/CookComputing.XmlRpcV2_dotnet45.dll && mv ../bin/CookComputing.XmlRpcV2.pdb ../bin/CookComputing.XmlRpcV2_dotnet45.pdb #building for dotnet4.5
```
Comment (kc284) [inline mk/dotnet-packages-build.sh]: Oops, here's something I missed the first time I looked at this PR: here you need a different run_msbuild targeting .NET 4.5.

**Comment by kc284** [comment]
Looks ok now. I'm merging it.

**Pull request merged**